### PR TITLE
Minor Seqexec Cleanup

### PR DESCRIPTION
**Pull request by tpolecat**
This PR removes the `ConfigDiff` abstraction and cleans up the `TcsController` type definitions a bit, removing unnecessary `boolean` members.

### CODE REVIEW

File: bundle/edu.gemini.seqexec.server/src/main/scala/edu/gemini/seqexec/server/TCS.scala [hunk 1]
```diff
@@ -22,21 +22,35 @@ import squants.space.{Millimeters, LengthConversions}
...
+    g2 // final result
 
   }
 
   private def guideOff(s0: TcsConfig, s1: Requested[TcsConfig]): SeqAction[Unit] =
-    TcsController.telescopeConfigDelta(s0.tc, s1.self.tc).offsetA match {
-      case Change(_, _) => tcsController.guide(GuideConfig(MountGuideOff, M1GuideOff, M2GuideOff))
-      case _            => tcsController.guide(computeGuideOff(s0, s1))
+    tcsController.guide {
+      if (s0.tc.offsetA == s1.self.tc.offsetA) computeGuideOff(s0, s1)
```
Comment (swalker2m) [inline bundle/edu.gemini.seqexec.server/src/main/scala/edu/gemini/seqexec/server/TCS.scala]: Perhaps we should define `Equal` instances for all the types.

Comment (tpolecat) [inline bundle/edu.gemini.seqexec.server/src/main/scala/edu/gemini/seqexec/server/TCS.scala]: Yep, good idea.

File: bundle/edu.gemini.seqexec.server/src/main/scala/edu/gemini/seqexec/server/TcsController.scala [hunk 2]
```diff
@@ -126,74 +132,57 @@ object TcsController {
...
-  object FollowOff extends FollowOption
-  object FollowOn extends FollowOption
-
-  sealed trait ProbeTrackingConfig {
-    def follow: FollowOption
-
-    def getNodChop: NodChopTrackingConfig
+  object FollowOption {
+    object FollowOff extends FollowOption
+    object FollowOn  extends FollowOption
```
Comment (swalker2m) [inline bundle/edu.gemini.seqexec.server/src/main/scala/edu/gemini/seqexec/server/TcsController.scala]: `case object`s here too?

Comment (tpolecat) [inline bundle/edu.gemini.seqexec.server/src/main/scala/edu/gemini/seqexec/server/TcsController.scala]: Yep, thanks.

File: bundle/edu.gemini.seqexec.server/src/main/scala/edu/gemini/seqexec/server/TcsControllerEpics.scala [hunk 3]
```diff
@@ -111,19 +113,26 @@ object TcsControllerEpics extends TcsController {
...
-          (bc, NodChop(Beam.B, Beam.C)), (ca, NodChop(Beam.C, Beam.A)), (cb, NodChop(Beam.C, Beam.B)),
-          (cc, NodChop(Beam.C, Beam.C))
-        ).foldLeft(Set[NodChop]())((s: Set[NodChop], v: (Boolean, NodChop)) => if (v._1) s + v._2 else s))
-      }
-      else NodChopTrackingConfig.None
-    }
-  }
+
+      // This last production is slightly tricky. 
+      o  <- if (List(aa, ab, ac, ba, bb, bc, ca, cb, cc).exists(identity)) {
```
Comment (swalker2m) [inline bundle/edu.gemini.seqexec.server/src/main/scala/edu/gemini/seqexec/server/TcsControllerEpics.scala]: Couldn't this be done better with a disjunction monoid?

**Comment by swalker2m** [comment]
Looks good. :+1:

File: bundle/edu.gemini.seqexec.server/src/main/scala/edu/gemini/seqexec/server/TcsControllerEpics.scala (hunk 3 above)
Comment (tpolecat) [inline bundle/edu.gemini.seqexec.server/src/main/scala/edu/gemini/seqexec/server/TcsControllerEpics.scala]: Yep, can make this better.

Comment (tpolecat) [inline bundle/edu.gemini.seqexec.server/src/main/scala/edu/gemini/seqexec/server/TcsControllerEpics.scala]: Actually I'm just going to replace these with `(_ == true)` etc. for now; I don't think the fold makes it any clearer.

**Pull request merged**